### PR TITLE
SW-106 add member repository test code & embedded mongodb

### PR DESCRIPTION
**Pull request by banggeunho**
[추가]
1. Test를 위한 embedded mongodb dependency 추가 (test scope 적용)
   `<dependency>
            <groupId>de.flapdoodle.embed</groupId>
            <artifactId>de.flapdoodle.embed.mongo</artifactId>
            <scope>test</scope>
        </dependency>`

2. test 환경 application.yaml 작성 (별도 전달)
3. MemberRepository Unit test 작성
4. RefreshTokenRepository Unit test 작성
5. DiaryRepository Unit test 예시 작성

[수정]
1. MemberRepository의 메소드 이름 변경 (findMembersByLoginId -> findByLoginId)
2. MemberRepository의 existsByLoginId ClassType 변경 (Optional<Members> -> boolean)

### CODE REVIEW

**Comment by banggeunho** [comment]
@KIM-JOOHYUN 
DiaryRepository unit test 예시로 조금 작성 해놨어요~
응용해서 테스트 진행하시면 됩니다.~
즐추~

**Comment by banggeunho** [comment]
test/resources/application.yml 생성하고,
아래 내용 추가 (테스트 환경에서 사용되는 application 설정정보입니다.)
------------------------------
spring:
  data:
    mongodb:
      database: test
      port: 0  # random port 지정

  main:
    allow-bean-definition-overriding: true

logging:
  level:
    org:
      springframework:
        boot:
          autoconfigure:
            mongo:
              embedded: DEBUG
    mongodb: DEBUG


home:
  url: http://localhost:3000/home/landing

jwt:
  header: Authorization
  secret: kasdisadjiadjioafjisdjfgiojgoijgioesrjtoesj449tje8ghd89r4tkgve9jg89ae4ute0ictj8g4htc34wr0
  access-token-expire-time: 1800000
  refresh-token-expire-time: 6048000




unsplash:
  api-key: qbG8uTxTAd4tQmBE59ox142Z8t2QktM9Lpi5TQXX3lY


kakao:
  auth:
    admin-key: b52388be69040492bc1fe0f9ba056e95
    login-redirect-url: http://localhost:3000?userId=
    logout-redirect-url: http://localhost:3000/home/landing
    logout_url: https://kapi.kakao.com
    token:
      url: https://kauth.kakao.com/oauth/token
      grant_type: authorization_code
      client_id: 6c41a2215aa9cbfe75475d959675d6a1
      redirect_url: http://localhost:8080/v0/members/kakao-login



    user-info:
      url: https://kapi.kakao.com/v2/user/me

  extract-keyword:
    url: https://c4cd6341-4bc1-4b4b-9c60-276492278f71.api.kr-central-1.kakaoi.io/ai/nlp/1da597890c064dcc9ed543908818b0f8
    api-key: 1ff6403660afde63511810bb5228c555

  translation:
    api-key: 629721b4b784e5e3a1a6975ec2beba54
    url: https://6876dc69-6900-465b-bec2-55f772b3c285.api.kr-central-1.kakaoi.io/ai/translation/231d835a1af54991bdbd02cf10de023e

  upload-storage:
    url: https://objectstorage.kr-central-1.kakaoi.io/v1/d21acfdf7b5b428a9127c5b267a4f3cd/web-hands-on

  download-storage:
    url: https://objectstorage.kr-central-1.kakaoi.io/v1/d21acfdf7b5b428a9127c5b267a4f3cd/resize-image

  emotion-analysis:
    url: https://a3d8fbea-c67e-4cac-8e14-f0af2ee1671f.api.kr-central-1.kakaoi.io/ai/conversation/d0bb985654284900a0dae2f99c2d2e4c
    api-key: e6ec925bef9e4de3a36a14cf356704b8

  api-token:
    url : https://iam.kakaoi.io/identity/v3/auth/tokens
    body: >
      {
          "auth": {
              "identity": {
                  "methods": [
                      "application_credential"
                  ],
                  "application_credential": {
                      "id": "bf022eaed0544dbca6a4da2c34464887",
                      "secret": "22t9h-P0poyugoCTsfUvobxz354bNM-N_4-uFIyKHAU-ACcFWOhWUBvqzEvzgXVUPNxF0tdcWRL7mWlVQeNdJA"
                  }
              }
          }
      }
--------------------

File: src/test/java/com/sweep/jaksim31/domain/diary/DiaryRepositoryTest.java [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+    @DisplayName("일기 전체 조회 (Read) - User ObjectId")
+    void findAllByUserId() {
+        //then
+        assertThat(diaryRepository.findAllByUserId("63c0cb6f30dc3d547e3b88bb").size())
+                .isEqualTo(20);
+    }
+
+    @Test
+    @Order(4)
+    @DisplayName("MongoTemplate")
```
Comment (KIM-JOOHYUN) [inline src/test/java/com/sweep/jaksim31/domain/diary/DiaryRepositoryTest.java:102]: MongoTemplate 테스트도 같이 할 수 있군요...!

File: src/test/java/com/sweep/jaksim31/domain/diary/DiaryRepositoryTest.java [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+import static org.junit.jupiter.api.Assertions.*;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.springframework.data.domain.Sort.by;
+
+@DataMongoTest
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+@TestExecutionListeners({DependencyInjectionTestExecutionListener.class, ExecutionTimeTestExecutionListener.class})
+@TestMethodOrder(value = MethodOrderer.OrderAnnotation.class) // 메소드 순서 지정
+class DiaryRepositoryTest {
```
Comment (KIM-JOOHYUN) [inline src/test/java/com/sweep/jaksim31/domain/diary/DiaryRepositoryTest.java:32]: Repository 테스트 코드 작성 감사합니다! 참고해서 필요한 테스트 코드 추가해보겠습니다:)

File: src/main/java/com/sweep/jaksim31/domain/members/MemberRepository.java [hunk 3]
```diff
@@ -17,11 +17,13 @@
...
  * 2023-01-11           김주현             조회 조건 수정 및 추가
  * 2023-01-18           김주현             id data type 변경(ObjectId -> String)
+ * 2023-01-21           방근호             existsByLoginId 메소드 type 변경(Optional<Members> -> boolean
+ * 2023-01-21           방근호             findMemberByLoginId 이름 변경 (-> findByLoginId)
  */
 
 public interface MemberRepository extends MongoRepository<Members, String> {
     Optional<Members> findById(String id);
-    Optional<Members> findMembersByLoginId(String loginId);
+    Optional<Members> findByLoginId(String loginId);
```
Comment (KIM-JOOHYUN) [inline src/main/java/com/sweep/jaksim31/domain/members/MemberRepository.java:26]: 더 간결하고 일관성 있어 보여서 더 좋은 것 같습니다! 수정 감사합니다:)

**Review by KIM-JOOHYUN — COMMENTED** [review]
테스트 코드 작성하시느라 정말 고생 많으셨습니다!  참고해서 Diary 테스트 수정해보겠습니다. 감사합니다:)
즐거운 설 연휴 보내세요!😊

**Pull request merged**